src/main/java/lotto/domain/Lotto.java

```diff
@@ -1,51 +1,53 @@
 package lotto.domain;
 
-import java.util.ArrayList;
-import java.util.HashSet;
 import java.util.List;
-import java.util.StringJoiner;
-import lotto.domain.vo.Number;
+import java.util.Set;
+import lotto.domain.vo.LottoNumber;
 
 public class Lotto {
 
     private static final int LOTTO_SIZE = 6;
 
-    private final List<Number> numbers;
+    private final Set<LottoNumber> lottoNumbers;
 
-    public Lotto(List<Number> numbers) {
-        validateSize(numbers);
-        validateDuplicateNumbers(numbers);
-        this.numbers = new ArrayList<>(numbers);
+    public Lotto(Set<LottoNumber> lottoNumbers) {
+        validateSize(lottoNumbers);
+        this.lottoNumbers = Set.copyOf(lottoNumbers);
+    }
```

Comment:
멀티 스레드 환경에서의 문제점을 얘기하지 못한 것 같아요.
validateSize를 통과하고 copyOf 하기 전에 다른 스레드가 lottoNumbers를 변경하면 도메인 규칙이 깨질 수 있어요

항상 방어적 복사가 먼저 선행되고 검증해야 thread-safe 해요 :)
```suggestion
    public Lotto(Set<LottoNumber> lottoNumbers) {
        lottoNumbers = Set.copyOf(lottoNumbers);
        validateSize(lottoNumbers);
        this.lottoNumbers = lottoNumbers
    }
```

Reply:
마침 오늘 이펙티브 자바 스터디에서 언급된 내용인데 반영해보겠습니다!!!